### ISSUE TITLE
NSLGameScanner.py fails to add game shortcuts when using separate prefix for each launcher

### ISSUE THREAD

**Issue by PseudoAnonMS** [moraroy/NonSteamLaunchers-On-Steam-Deck#271]
After starting fresh, and selecting to use separate prefixes, the script is failing to add game shortcuts into steam.
It was working before when I did not have separate prefixes. The launcher shortcut itself was added and is working.
Here are the logs for the script:
> ['/home/deck/.config/systemd/user/Modules', '/home/deck/.config/systemd/user', '/usr/lib/python311.zip', '/usr/lib/python3.11', '/usr/lib/python3.11/lib-dynload', '/usr/lib/python3.11/site-packages']
['/home/deck/.config/systemd/user/Modules', '/home/deck/.config/systemd/user/Modules', '/home/deck/.config/systemd/user', '/usr/lib/python311.zip', '/usr/lib/python3.11', '/usr/lib/python3.11/lib-dynload', '/usr/lib/python3.11/site-packages']
Service started.
['/home/deck/Downloads/NonSteamLaunchersInstallation/lib/python3.11/site-packages', '/home/deck/.config/systemd/user/Modules', '/home/deck/.config/systemd/user/Modules', '/home/deck/.config/systemd/user', '/usr/lib/python311.zip', '/usr/lib/python3.11', '/usr/lib/python3.11/lib-dynload', '/usr/lib/python3.11/site-packages']
Epic Games is not installed. Skipping.
Existing shortcut found based on shortcut ID for game Gog Galaxy. Skipping creation.
Updated CompatToolMapping entry for appid: 3216372511
Ubisoft Connect is not installed. Skipping.
Battle.net is not installed. Skipping.
EA App is not installed. Skipping.
Amazon Games is not installed. Skipping.
itch.io is not installed. Skipping.
Legacy Games is not installed. Skipping.
Humble Bundle is not installed. Skipping.
IndieGala Client is not installed. Skipping.
Rockstar Games Launcher is not installed. Skipping.
Glyph is not installed. Skipping.
Minecraft: Java Edition is not installed. Skipping.
Playstation Plus is not installed. Skipping.
VK Play is not installed. Skipping.
Xbox Game Pass is not installed. Skipping.
GeForce Now is not installed. Skipping.
Netflix is not installed. Skipping.
Hulu is not installed. Skipping.
Disney+ is not installed. Skipping.
Amazon Prime Video is not installed. Skipping.
Youtube is not installed. Skipping.
Amazon Luna is not installed. Skipping.
Twitch is not installed. Skipping.
movie-web is not installed. Skipping.
Fortnite is not installed. Skipping.
The app ID for Gog Galaxy is 3216372511
Epic Games: EpicGamesLauncher does not exist
Gog Galaxy: GogGalaxyLauncher exists
Traceback (most recent call last):
  File "/home/deck/.config/systemd/user/NSLGameScanner.py", line 525, in <module>
    os.rename(current_path, new_path)
IsADirectoryError: [Errno 21] Is a directory: '/home/deck/.local/share/Steam/steamapps/compatdata/GogGalaxyLauncher' -> '/home/deck/.local/share/Steam/steamapps/compatdata/3216372511'

**Comment by moraroy** [comment]
@PseudoAnonMS this still needs to be fixed I believe

**Comment by Neoxon4865** [comment]
I also used the prefix thing to install Battle.net. It is failing to detect diablo 4 as a shortcut which unfortunately is a bummer

**Comment by moraroy** [comment]
@Neoxon4865 yeah that scanner doesn't have full SD card support yet...meaning it will detect your games if you DONT use the separate prefixes still working on the scanners slowly but surely. It should pick it up if you choose one prefix.

**Comment by Neoxon4865** [comment]
> @Neoxon4865 yeah that scanner doesn't have full SD card support yet...meaning it will detect your games if you DONT use the separate prefixes still working on the scanners slowly but surely. It should pick it up if you choose one prefix.

Yeah I was having issue trouble getting steam deck to recognize the battle net launcher on steam OS (steam launcher in window recognized it well). So I tried to reinstalled it using the separate app id check box and it finally recognize it under steam OS under non steam games. So maybe if I reinstall battle net as non-separate app id, albeit it hasn’t work. But I am hoping this will get resolve soon but so far I only use battle net for diablo 4 so it doesn’t matter much.

**Comment by moraroy** [comment]
@Neoxon4865 when did you try this installing battle net? as of today the launchers seem to all be installing and putting shortcuts in steam, are you using the Desktop Version or the Decky Plugin version? as far as it finding games...it should find the games if you have installed them in the default location of Battlenet, i havnt made it to look for other areas in the system yet.

**Comment by Neoxon4865** [comment]
@moraroy So I got my steam deck today (yay!) and I did start installing Battle.net thru your program in desktop mode. It wasn’t appearing at all as a desktop shortcut until I did the seperate app ID trick I found in another issues. What I do is launch battle.net thru steam in Game Mode, then I pressed to install diablo 4. Then it still doesn’t show up after launching the game and restarting the steam multiple time, and also launching non steam launcher to restart the auto finding service. I could try the decky version but tbh downloading 50gbs of diablo 4 3 times today sucked so I will hear your input on what I could do better or did wrong.

**Comment by moraroy** [comment]
@Neoxon4865 you didn't do anything wrong brother I don't have that game specifically..but I do have overwatch I'll re install and see what's going on, in the mean time , on this github repo look at the NSLGameScanner.py file. It will have the battle net scanner in there.

**Comment by moraroy** [comment]
@Neoxon4865 you also don't need to reinstall anything to use the plugin! They work togather!
